### PR TITLE
Fix flows on OBCluster; update docs for 2.1.2 release; replace log level(-2) with TRACE 

### LINKED ISSUE #179: [Bug]: Can not pull image within bootstrap timeout when creating obcluster at the first time
### Describe the bug

With poor network conditions, it could be difficult to pull oceanbase image before the progress reaches bootstrap timeout. This leads to failure of the first deployment of oceanbase cluster on a worker node at almost all cases, which could be struggling for new-coming users.

### Environment

Poor networking access to image registry.

### Fast reproduce steps

1. Choose a new cluster where oceanbase images have not been pulled as the reproducing environment.
2. Follow the quick start in README.md, install cert-manager, install ob-operator, create users' secrets and create obcluster
3. The cluster will get stuck in `wait bootstrap` status and print error message on and on.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

1. Wait for the image to be pulled down first, fix #179
2. Validate standalone mode before creating obzones and observer
3. Update docs for 2.1.2 release
4. Replace LEVEL(-2) with TRACE when start with log-vebosity=2

### CODE REVIEW

File: example/quickstart/obcluster.yaml [hunk 1]
```diff
@@ -9,9 +9,6 @@ spec:
   serviceAccount: "default"
   userSecrets:
     root: root-password
-    proxyro: proxyro-password
-    monitor: monitor-password
-    operator: operator-password
```
Comment (chris-sun-star) [inline example/quickstart/obcluster.yaml:14]: also update observer's image

File: example/quickstart/tenant.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apiVersion: oceanbase.oceanbase.com/v1alpha1
+kind: OBTenant
+metadata:
+  name: t1
```
Comment (chris-sun-star) [inline example/quickstart/tenant.yaml:4]: maybe a more meaningful name

File: internal/resource/obcluster/obcluster_task.go [hunk 3]
```diff
@@ -1029,3 +978,158 @@ outer:
 	}
 	return nil
 }
+
+func (m *OBClusterManager) CheckImagePullReady() tasktypes.TaskError {
```
Comment (chris-sun-star) [inline internal/resource/obcluster/obcluster_task.go:982]: CheckImageReady

File: example/quickstart/obcluster.yaml (hunk 1 above)
Comment (PowerfooI) [reply example/quickstart/obcluster.yaml:14]: Shall we update to the latest `4.2.1.3-103000032023122818`?

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**